app/GroupTask.php

```diff
@@ -153,22 +186,51 @@ public static function initializeCryptoTasks($group_id, $randomize,$final) {
       $taskArray = '[
       {"taskName":"Cryptography","taskParams":{"hasIndividuals":"false","intro":"group_1","hasGroup":"true","mapping":"random","maxResponses":"10","type":"intro"}},
       {"taskName":"Cryptography","taskParams":{"hasIndividuals":"false","intro":"group_1","hasGroup":"true","mapping":"random","maxResponses":"10","type":"task"}},
-      {"taskName":"GroupSurvey","taskParams":{"hasIndividuals":"false","hasGroup":"true","type":"1","statementOrder":"ordered"}},
+      {"taskName":"GroupSurvey","taskParams":{"hasIndividuals":"false","hasGroup":"true","type":"1","statementOrder":"ordered","task_name":"Cryptography"}},
       {"taskName":"Conclusion","taskParams":{"hasIndividuals":"true","hasGroup":"false","type":"mturk","hasCode":"false","displayScoreGroup":"false", "digitalReceipt":"false", "sonaId": "547", "payment": "30", "feedback":"true", "feedbackLinkType":"qualtrics"}}
       ]';
     }
     else{
       $taskArray = '[
       {"taskName":"Cryptography","taskParams":{"hasIndividuals":"false","intro":"group_1","hasGroup":"true","mapping":"random","maxResponses":"10","type":"intro"}},
       {"taskName":"Cryptography","taskParams":{"hasIndividuals":"false","intro":"group_1","hasGroup":"true","mapping":"random","maxResponses":"10","type":"task"}},
-      {"taskName":"GroupSurvey","taskParams":{"hasIndividuals":"false","hasGroup":"true","type":"1","statementOrder":"ordered"}},
+      {"taskName":"GroupSurvey","taskParams":{"hasIndividuals":"false","hasGroup":"true","type":"1","statementOrder":"ordered","task_name":"Cryptography"}},
       {"taskName":"WaitingRoom","taskParams":{"hasIndividuals":"false","hasGroup":"true","task":"1"}}
       ]';
     }
     
     return Self::initializeTasks($group_id, $taskArray, $randomize);
   }
 
+  public static function initializeMemoryTasks($group_id, $randomize,$final,$test) {
+    if($final){
+      $taskArray = '[
```

Comment:
This is a major holdover from Gabe's design that I chose to integrate rather than rebuild. I think if I had to justify why he did it this way, it's because we have generalized tables for GroupTasks and IndividualTasks, all of which have different subtypes (for instance, cryptography vs memory), and each subtype has some parameters that don't overlap with the other subtypes. So I think in order to avoid a prohibitively wide table and to keep these parameters accessible in a generalized way, he stringified these arrays so he could store them in one column called "parameters"

Reply:
I see okay, thanks for clarifying. Figured that this was a holdover but was curious about it